crates/typst-library/src/meta/heading.rs

```diff
@@ -120,6 +120,22 @@ pub struct HeadingElem {
     #[default(Smart::Auto)]
     pub bookmarked: Smart<bool>,
 
+    /// The heading's title appearing in the outline, and maybe in the
+    /// bookmarks.
+    ///
+    /// The default value of `{auto}` indicates that the title will be the same
+    /// as `body`.
+    #[default(Smart::Auto)]
+    pub outline_title: Smart<Content>,
+
+    /// The heading's title appearing in the bookmarks.
+    ///
+    /// The default value of `{auto}` indicates that the title will be the same
+    /// as `outline-title`. If `outline-title` is also `{auto}`, the title will
+    /// be the same as `body`.
+    #[default(Smart::Auto)]
+    pub bookmark_title: Smart<Content>,
```

Comment:
This should probably be a string, since it's going to be stringified in the end anyway as it's PDF metadata.

Reply:
I wonder whether we need it at all. Shouldn't `outline_title` suffice for pretty much every use case? I think adding too many knobs isn't good either.

Reply:
I see that it would be parallel with `bookmarked`. I can't say that I'm a big fan of that existing either though. But I see how it can be useful, which I'm having a harder time for here. I think a `pdf.boomark` function will exists sooner or later (in a `pdf` module. and there will be a `html` module for html primitives), which will allow arbitrary customization.

Reply:
I think it's useful to display something different for the PDF metadata, as it's a string, and content to string conversion isn't always lossless.

I should note that this could very well be reused for something like HTML or EPUB or anything that would take a stringified table of contents.

Reply:
@PgBiel I am actually considering to allow content for `set document(title)` because it is such a common pitfall. Together with get rules this would allow to move more general properties from the template function to document set rules (also author e.g. which could allow richer standardized metadata like email and institution). For PDF, we would then use the internal content to plain-text functionality that is already used for bookmarks. If we indeed go down that route, it would be strange to introduce further string-only argument for the sake of PDF.

Reply:
The benefit of standardization for such metadata of course being that it's easier to switch templates.

Reply:
Why would you have a footnote in the outline heading? The example seems pretty contrived to me to be honest.

Reply:
Well... Another example would be having manual linebreaks in the outline but not in the bookmark (linebreaks in `content` result in spaces when converted to `string`, which is not expected in Chinese).

Reply:
> I should note that this could very well be reused for something like HTML or EPUB or anything that would take a stringified table of contents.

Then, the name `bookmark-title` would be super confusing if there's also `outline-title`.